database/src/pool/postgres.rs

```diff
@@ -1868,25 +1875,101 @@ where
             None => Ok(None),
             Some(row) => {
                 let job = BenchmarkJob {
+                    id: row.get::<_, i32>(0) as u32,
                     target: *target,
-                    backend: CodegenBackend::from_str(&row.get::<_, String>(0))
+                    backend: CodegenBackend::from_str(&row.get::<_, String>(1))
                         .map_err(|e| anyhow::anyhow!(e))?,
-                    profile: Profile::from_str(&row.get::<_, String>(1))
+                    profile: Profile::from_str(&row.get::<_, String>(2))
                         .map_err(|e| anyhow::anyhow!(e))?,
-                    request_tag: row.get::<_, String>(2),
+                    request_tag: row.get::<_, String>(3),
                     benchmark_set: benchmark_set.clone(),
-                    created_at: row.get::<_, DateTime<Utc>>(3),
+                    created_at: row.get::<_, DateTime<Utc>>(4),
                     // The job is now in an in_progress state
                     status: BenchmarkJobStatus::InProgress {
-                        started_at: row.get::<_, DateTime<Utc>>(4),
+                        started_at: row.get::<_, DateTime<Utc>>(5),
                         collector_name: collector_name.into(),
                     },
-                    retry: row.get::<_, i32>(5) as u32,
+                    retry: row.get::<_, i32>(6) as u32,
                 };
                 Ok(Some(job))
             }
         }
     }
+
+    async fn mark_benchmark_request_as_completed(&self, tag: &str) -> anyhow::Result<bool> {
+        // Find if the benchmark is completed and update it's status to completed
+        // in one SQL block
+        let row = self
+            .conn()
+            .query_opt(
+                "
+                UPDATE
+                    benchmark_request
+                SET
+                    status = $1,
+                    completed_at = NOW()
+                WHERE
+                    benchmark_request.tag = $2
+                    AND benchmark_request.status != $1
+                    AND NOT EXISTS (
+                        SELECT
+                            1
+                        FROM
+                            job_queue
+                        WHERE
+                            request_tag = benchmark_request.tag
```

Comment:
Let's use `job_queue.request_tag` and `job_queue.status` here. To be consistent with the subquery below, but mainly to avoid using attributes of the parent query by accident.

Reply:
c9b32be9cefd809e408509e549adfcc73c4984f0